src/blocks/weather/nws.rs

```diff
@@ -0,0 +1,391 @@
+    }
+
+    fn icon_to_word(icon: WeatherIcon) -> String {
+        match icon {
+            WeatherIcon::Clear { .. } => "Clear",
+            WeatherIcon::Clouds { .. } => "Clouds",
+            WeatherIcon::Fog { .. } => "Fog",
+            WeatherIcon::Thunder { .. } => "Thunder",
+            WeatherIcon::Rain { .. } => "Rain",
+            WeatherIcon::Snow => "Snow",
+            WeatherIcon::Default => "Unknown",
+        }
+        .to_string()
+    }
+
+    fn wind_kmh(&self) -> f64 {
+        if self.wind_speed.unit_code.ends_with("km_h-1") {
+            self.wind_speed.value
+        } else {
+            self.wind_speed.value * MPH_TO_KPH
+        }
+    }
+
+    fn apparent_temp(&self) -> f64 {
+        let temp = if self.temperature.unit_code.ends_with("degC") {
+            self.temperature.value
+        } else {
+            (self.temperature.value - 32.0) * 5.0 / 9.0
+        };
+        let humidity = self.relative_humidity.value;
+        let wind_speed = self.wind_kmh();
+        australian_apparent_temp(temp, humidity, wind_speed)
+    }
+
+    fn to_moment(&self) -> WeatherMoment {
+        let icon = short_forecast_to_icon(&self.short_forecast, !self.is_daytime);
+        let weather = Self::icon_to_word(icon);
+        WeatherMoment {
+            icon: short_forecast_to_icon(&self.short_forecast, !self.is_daytime),
+            weather,
+            weather_verbose: self.short_forecast.clone(),
+            temp: self.temperature.value,
+            apparent: self.apparent_temp(),
+            humidity: self.relative_humidity.value,
+            wind: self.wind_speed.value,
+            wind_kmh: self.wind_kmh(),
+            wind_direction: Some(self.wind_direction()),
+        }
+    }
+
+    fn to_aggregate(&self) -> ForecastAggregate {
+        ForecastAggregate {
+            temp: self.temperature.value,
+            apparent: self.apparent_temp(),
+            humidity: self.relative_humidity.value,
+            wind: self.wind_speed.value,
+            wind_kmh: self.wind_kmh(),
+            wind_direction: Some(self.wind_direction()),
+        }
+    }
+}
+
+fn aggregate_avg(data: &[ForecastAggregate]) -> ForecastAggregate {
+    let mut temp = 0.0;
+    let mut apparent = 0.0;
+    let mut humidity = 0.0;
+    let mut wind_north = 0.0;
+    let mut wind_east = 0.0;
+    let mut wind_kmh_north = 0.0;
+    let mut wind_kmh_east = 0.0;
+    for val in data {
+        temp += val.temp;
+        apparent += val.apparent;
+        humidity += val.humidity;
+        let (sin, cos) = val
+            .wind_direction
+            .unwrap_or_default()
+            .to_radians()
+            .sin_cos();
+        wind_north += val.wind * cos;
+        wind_east += val.wind * sin;
+        wind_kmh_north += val.wind_kmh * cos;
+        wind_kmh_east += val.wind_kmh * sin;
+    }
+
+    let count = data.len() as f64;
+    ForecastAggregate {
+        temp,
+        apparent,
+        humidity,
```

Comment:
Temp, apparent, and humidity should be divided by count.

Reply:
Total mistake on my part; fixed.